mint-001.md

```diff
@@ -0,0 +1,182 @@
+# mint-001
+
+## 3 Key Degrading Multisig
+
+## Motivation
+
+The 2-of-3 multisig serves as a standard in multisignature solutions, offering a well-balanced combination of redundancy, security, flexibility, and practicality for securing Bitcoin. Loss of a single key does not spell disaster, and key distribution enhances security. Miniscript's time feature enables further disaster recovery. If two keys are lost, a timelock can convert the wallet to a 1-of-3 multisig. The proposal outlines four conditions through Miniscript's `thresh()`:
+
+### 2 Layer Conditions to be satisfied using:
+
+#### Base expressions (type B).
+
+##### timelock: **relative** or **absolute** [^timelock] [^either]
+
+- **older(**int**)** [^older]
+
+- **after(**int**)** [^after]
+
+#### Key expressions[^k_type] (type K).
+
+- **pk(**$Key_1$**)**
+
+- **pk(**$Key_2$**)**
+
+- **pk(**$Key_3$**)**
+
+Initially, the wallet requires 2-of-3 keys, functioning as a traditional 2-of-3 multisig. After the timelock expires, only one key is needed to spend the funds.
+
+### More on Timelock Values
+
+- Descriptor template security is independent of relative or absolute timelock values; the impact is on timelock duration only. Reference transactions on testnet use short-duration timelocks; however, practical applications will vary.
+
+- Relative timelock descriptors provide a persistent structure, requiring a self-send to extend timelock security, thus enabling standardized timelock durations.
+
+- Absolute timelock descriptors need frequent updates; setting long-term timelock values becomes cumbersome. Updating these descriptors entails transferring coins to a new descriptor with revised absolute timelock values.
+
+##### Suggested Relative Block Height Timelocks:
+
+- **older(**32800**)** - Halfway point of block height relative timelock [^278days]
+
+- **older(**65535**)** - Maximum duration of a block height relative timelock [^455days]
+
+##### Suggested Relative Epoch Timmelocks:
+
+- **older(**4224680**)** - Approximate Halfway point of epoch time relative timelock[^180days]
+
+- **older(**4259839**)** -- Maximum duration of an epoch time relative timelock[^388days]
+
+##### Suggested Relative Epoch Timmelocks: older(4224679)
+
+- Mid-point epoch time relative timelock (\~180 days, 6 months) older(4259839)
+
+- Max duration epoch time relative timelock (\~388 days)
+
+Below is a reference diagram on how the 3 Key Time Layered Multisig operates across time:
```

Comment:
In ab8f15bdfc9751d373f29da38a491d8948c4df29:

I'm not sure where this should go, but somewhere in this doc we should say that the first timelock should be less than the second. This doesn't matter at all semantically but it's good to tighten the spec wherever we can, and it'd be potentially confusing to have the timelocks in an arbitrary order.

Reply:
For clarity please provide a few examples that I can compile at:

https://bitcoin.sipa.be/miniscript

So I can better understand your intentions.

I will be able to incorporate a "Suggested Best Practices" into the docs.

cc: @Rob1Ham

Reply:
Mis-read your comment - no need for examples.

Reply:
https://github.com/Blockstream/miniscript-templates/pull/11#discussion_r1443220787

Reply:
I have been using the term "Cascading Timelocks" in some of my own template proposals.
We are on the same page with this!